Source/multi.cpp

```diff
@@ -747,17 +743,6 @@ BOOL NetInit(BOOL bSinglePlayer, BOOL *pfExitProgram)
 		sgGameInitInfo.bTheoQuest = sgOptions.Gameplay.bTheoQuest;
 		sgGameInitInfo.bCowQuest = sgOptions.Gameplay.bCowQuest;
 		sgGameInitInfo.bFriendlyFire = sgOptions.Gameplay.bFriendlyFire;
-		memset(&ProgramData, 0, sizeof(ProgramData));
-		ProgramData.size = sizeof(ProgramData);
-		ProgramData.maxplayers = MAX_PLRS;
-		ProgramData.initdata = &sgGameInitInfo;
-		memset(&plrdata, 0, sizeof(plrdata));
-		plrdata.size = sizeof(plrdata);
-		memset(&UiData, 0, sizeof(UiData));
-		UiData.size = sizeof(UiData);
-		UiData.selectnamecallback = mainmenu_select_hero_dialog;
-		UiData.changenamecallback = (void (*)())mainmenu_change_name;
-		UiData.profilefields = UiProfileGetString();

```

Comment:
UiData originally served as a way to send function pointers to DiabloUI.dll, but since we statically link them this isn't needed and we can simply call things direction.
Doing so helped to reveal that a lot of things were never used which then resulted in the other big deletions.

Some of the structs here have been shaved down a bit since we do not interact with BNet and so do not need to provide it with stats and other non-essential metadata, so this PR just removes the last bits after making sure that they are not needed for anything we do.